src/main/java/com/iexec/core/task/Task.java

```diff
@@ -178,4 +178,19 @@ public boolean inCompletionPhase() {
     public boolean isTeeTask() {
         return TeeUtils.isTeeTag(getTag());
     }
+
+    /**
+     * A task is eligible to the Contribute And Finalize flow
+     * if it matches the following conditions:
+     * <ul>
+     *     <li>It is a TEE task
+     *     <li>Its trust is 1
+     * </ul>
+     *
+     * @return {@literal true} if eligible, {@literal false} otherwise.
+     */
+    public boolean isEligibleToContributeAndFinalize() {
+        return isTeeTask()
+                && trust == 1;
```

Comment:
There is a difference with iexec-commons-poco where we check if there is a callback as well.

Reply:
Wouldn't it be better to implement the same check ?